File: navigation/src/main/java/com/example/navigation/stepsEngine/field/PinTypeField.java
package com.example.navigation.stepsEngine.field;

import com.example.navigation.stepsEngine.payment.FlowState;

public class PinTypeField extends Field {

    @Override
    public String get(FlowState flowState) {
        return flowState.getCard().getPin().getPinType().name();
    }

    @Override
    public String getId() {
        return null;
    }

    @Override
    public void set(FlowState flowState, Object value) {

    }

}

File: navigation/src/main/java/com/example/navigation/stepsEngine/field/PinTypeField.kt
package com.example.navigation.stepsEngine.field

import com.example.navigation.stepsEngine.payment.FlowState
import kotlinx.android.parcel.Parcelize

@Parcelize
class PinTypeField : Field() {
    override operator fun get(flowState: FlowState): String? {
        return flowState.card.pin.pinType?.name
    }

    override fun getId(): String {
        return FieldName.PIN_TYPE
    }

    override operator fun set(flowState: FlowState, value: Any?) {}
}